File: DotNet/nullCoalescing.cs
// C# - Null Coalescing Operator
string suppliedName = "Tyler Durden";
string? narratorName;

// Coalescing
string name = suppliedName ?? "Jack";
string name2 = narratorName ?? "Jack";

Console.WriteLine(name) // Output: "Tyler Durden"
Console.WriteLine(name2) // Output: "Jack"

/*
* Facebook: https://www.facebook.com/archcode01/photos/a.101228608227174/101244494892252/?type=3&theater
* Instagram: https://www.instagram.com/p/B-xigZ4AuKE/
* Twitter: https://twitter.com/arch_code/status/1251549335394947073/photo/1
*/

File: PHP/nullCoalescing.php
<?php
    // PHP - Null coalescing operator
    $array = ['one', 'two'];

    // Ternary
    $one = isset($array[0]) ? $array[0] : 'empty'; // value exist

    $three = isset($array[2]) ? $array[2] : 'empty'; // value doesn't exist
    

    // Coalescing operator
    $one = $array[0] ?? 'empty'; // value exist

    $three = $array[2] ?? 'empty'; // value doesn't exist

    /*
    * Facebook: https://www.facebook.com/archcode01/photos/a.101228608227174/101239041559464/?type=3&theater
    * Instagram: https://www.instagram.com/p/B-uH3gWgLQb/
    * Twitter: https://twitter.com/arch_code/status/1251547603533266946/photo/1
    */
?>